lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -986,7 +1042,11 @@ bool TypeSystemSwiftTypeRef::Verify(opaque_compiler_type_t type) {
 
 #include <regex>
 namespace {
-template <typename T> bool Equivalent(T l, T r) { return l == r; }
+template <typename T> bool Equivalent(T l, T r) {
+  if (l != r)
+    llvm::dbgs() <<  l << " != " << r << "\n";
```

Comment:
This odd function is used by the assertion that is part of the VALIDATE_AND_RETURN macro, that runs each query through TypeSystemSwiftTypeRef and SwiftASTContext and compares the results to make sure the new implementation doesn't diverge from the old one without us knowing and deciding that the divergence is acceptable. Since it's only run inside of assert() we might as well dump useful debug information in the failure case.
The idea is to delete the SwiftASTContext functions once the transition is complete and then this function in all its ugliness will be deleted, too.